lib/string.js

```diff
@@ -430,4 +437,31 @@ internals.String.prototype.trim = function () {
     return obj;
 };
 
+
+internals.String.prototype.replace = function (pattern, replacement) {
+
+    Hoek.assert(pattern instanceof RegExp, 'pattern must be a RegExp');
+    Hoek.assert(typeof replacement === 'string', 'replacement must be a String');
+
+    var obj = this._test('replace', undefined, function (value, state, options) {
+
+        if (options.convert) { // we do NOT check the pattern if not converting, that's the job of "regex(...)"
+            return null;
+        }
+
+        return Errors.create('string.replace', { value: value }, state, options);
+    });
+
+    if (!obj._inner.replacements) {
+        obj._inner.replacements = [];
+    }
+
+    obj._inner.replacements.push(function (value) {
```

Comment:
Do not push functions, push parameters that will be used directly in the loop up there without function calls.